OpenXava/src/org/openxava/web/editors/AmazonS3Persistor.java

```diff
@@ -0,0 +1,158 @@
+package org.openxava.web.editors;
+
+import java.io.*;
+import java.lang.reflect.*;
+import java.util.*;
+
+import org.apache.commons.io.*;
+import org.apache.commons.logging.*;
+import org.openxava.util.*;
+
+import com.openxava.naviox.util.*;
+
+public class AmazonS3Persistor {
+	
+   private static final java.io.File PARENT  = new java.io.File(XavaPreferences.getInstance().getFilesPath());
+   private static final String PrefixFolderName = NaviOXPreferences.getInstance().getAmazonBucketPrefixFolderName();
+   private static Log log = LogFactory.getLog(AmazonS3Persistor.class); 
+   
+   public static Object initialize() {
+	   
+	   /** 
+	    * Amazon s3 credentials 
+	    */
+	   String accessKey = NaviOXPreferences.getInstance().getAmazonAccessKey();
+	   
+	   String secretKey = NaviOXPreferences.getInstance().getAmazonSecretKey();	   
+	   
+	   String region = NaviOXPreferences.getInstance().getAmazonRegion();
+	   try {
+	       
+	       Class<?> awsClientBuilder = Class.forName("com.amazonaws.client.builder.AwsClientBuilder");
+	       Object result = Class.forName("com.amazonaws.services.s3.AmazonS3ClientBuilder")
+	       						.getMethod("standard")
+	       						.invoke(null);
+	       
+	       if (!secretKey.isEmpty() && !accessKey.isEmpty()) {
+	    	   Object credentials = Class.forName("com.amazonaws.auth.BasicAWSCredentials")
+		   				.getConstructor(String.class, String.class)
+  						.newInstance(accessKey, secretKey);
+		       
+	    	   Object awsStaticCredentialsProvider = Class.forName("com.amazonaws.auth.AWSStaticCredentialsProvider")
+						.getConstructor(Class.forName("com.amazonaws.auth.AWSCredentials"))
+						.newInstance(credentials);
+		       
+	    	   result = awsClientBuilder.getMethod("withCredentials", Class.forName("com.amazonaws.auth.AWSCredentialsProvider"))
+	    			   					   .invoke(result, awsStaticCredentialsProvider);
+	    	   
+	    	   result = awsClientBuilder.getMethod("withRegion", String.class).invoke(result, region);
+	       }
+	       
+	       return awsClientBuilder.getMethod("build").invoke(result);
+	        
+	   } catch(Exception ex) {
+		   
+		 ex.printStackTrace();
```

Comment:
This needs to throw proper meaningful runtimeexception to the user. Perhaps each AWS exception should be translated to a different useful error message